tests/ui/coroutine/ref-escapes-but-not-over-yield.rs

```diff
@@ -1,16 +1,13 @@
 #![feature(coroutines, stmt_expr_attributes)]
 
 fn foo(x: &i32) {
-    // In this case, a reference to `b` escapes the coroutine, but not
-    // because of a yield. We see that there is no yield in the scope of
-    // `b` and give the more generic error message.
-    let mut a = &3;
+    let a = &mut &3;
```

Comment:
why did you change this test? if this changes observable behavior, then it probably needs an FCP at least.

Reply:
I think so. We might need one.

On a closer look, if I understand this test correctly, it does not seem to test  the right thing? `a: &i32` is moved into the coroutine `b` and it is assigned to another borrowed `i32` with a shorter lifetime, which should have been allowed. Relocating upvars with this PR actually let this original test to work, which is really confusing.

Judging by the intended error message, I am proposing a more appropriate test as laid out in this patch.